sitehost/server/resource.go

```diff
@@ -0,0 +1,155 @@
+// Package server provides the functions to create a Server resource via SiteHost API.
+package server
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/sitehostnz/gosh"
+	"github.com/sitehostnz/terraform-provider-sitehost/sitehost/helper"
+)
+
+// Resource returns a schema with the operations for Server resource.
+func Resource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: createResource,
+		ReadContext:   readResource,
+		UpdateContext: updateResource,
+		DeleteContext: deleteResource,
+
+		Schema: resourceSchema,
+	}
+}
+
+// createResource function to create a new Server resource.
+func createResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*helper.CombinedConfig).Client
+	keys := d.Get("ssh_keys").([]interface{})
+
+	sshKeys := make([]string, 0, len(keys))
+	for _, key := range keys {
+		sshKeys = append(sshKeys, key.(string))
+	}
+
+	opts := &gosh.ServerCreateRequest{
+		Label:       d.Get("label").(string),
+		Location:    d.Get("location").(string),
+		ProductCode: d.Get("product_code").(string),
+		Image:       d.Get("image").(string),
+		Params: gosh.ParamsOptions{
+			SSHKeys: sshKeys,
+		},
+	}
+
+	res, err := client.Servers.Create(ctx, opts)
+	if err != nil {
+		return diag.Errorf("Error creating server: %s", err)
+	}
+
+	// Set data
+	d.SetId(res.Return.Name)
+	err = d.Set("name", res.Return.Name)
```

Comment:
same here

Reply:
Could you let me more info about what means that comment pl, because the previous comment was related to return just d.Set...